pages/_app.js

```diff
@@ -1,11 +1,45 @@
 import GlobalStyle from "../styles";
 import Layout from "@/components/Layout/Layout";
+import { useState } from "react";
+import initialFlashcards from "@/assets/flashcards.json";
+import collections from "@/assets/collections.json";
+import useLocalStorageState from "use-local-storage-state";
 
 export default function App({ Component, pageProps }) {
+  const [flashcards, setFlashcards] = useLocalStorageState("flashcards", {
+    defaultValue: initialFlashcards,
+  });
+
+  function handleIsCorrect(id) {
+    setFlashcards(
+      flashcards.map((flashcard) => {
+        return flashcard.id === id
+          ? { ...flashcard, isCorrect: !flashcard.isCorrect }
+          : flashcard;
+      })
```

Comment:
when you update this flashcards array, the entire list of flashcards are rendered again, therefore also the card where the state was toggled. This means you don't need an internal "isCorrect" state, since the compnent is rerendered with the new value given by the prop "flashcard".